src/server/node_manager.c

```diff
@@ -8776,3 +8779,57 @@ update_node_rassn(attribute *pexech, enum batch_op op)
 {
 	update_job_node_rassn(NULL, pexech, op);
 }
+
+static void
+set_resv_for_degrade(struct pbsnode *pnode, resc_resv *presv)
+{
+	char *str_time;
+	long degraded_time;
+
+	if (presv->ri_wattr[RESV_ATR_resv_standing].at_val.at_long == 0)
+		presv->ri_degraded_time = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
```

Comment:
i have kept the logic same as in vnode_unavailable()

Reply:
I guess I don't understand what presv->ri_degraded_time is used for.  I was assuming it was the time the reservation entered the degraded state.  Speaking of vnode_unavailable(), is there any chance you can call your new function from there, or do they do different things?

Reply:
ri_degraded_time is used as the soonest time in the reservation window when the reservation becomes degraded. For a standing reservation, it could be that an instance after the first is degraded. Using ri_degraded_time, reserve_retry_cutoff and reserve_retry_init, we find the retry time.
when a node goes down, vnode_unavailable() is called. In that function we look for all the reservations associated with the vnode that went down, set the degraded time and calculate the retry time somewhere between now and the degraded time.
Over here, we know the reservation as well as the node that went down, we need not "find" the reservation and also not go through the logic of finding things which we already know of.

Reply:
I wasn't thinking a wholesale replacement of vnode_unavailable().  I was thinking that this function handles what should happen if a reservation goes degraded.  So, do what you need to do in vnode_unavailable() to find the reservations, and then call your new function.  I guess I'm not seeing something.  If it doesn't make sense, then don't do it.  I was just hoping for some code reuse.

Reply:
vnode_unavailable() relies on what is there in ri_pbsnode_list to find the reservations. We set ri_pbsnode_list in set_nodes() once everything goes fine.
vnode_unavailable() is called when a node goes down. So, it was called when the scheduling cycle was in progress and the node went down. But, at that time "this" reservation was not updated with "this" node. Even if we call vnode_unavailable() from here, it will not help.